### PR TITLE
Refactoring of Module Loading API

### PR DESCRIPTION
**Pull request by david-waltermire**
# Committer Notes

This PR is the result of a significant refactor of the module loading and data binding layers of the library. The result is a much cleaner API that should perform better overall, due to shorter code paths. This version is much more expansible and enabled dynamic compilation of module classes to be disabled by default. Some of these changes are backwards compatibility breaking and will likely require a major release.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/metaschema-framework/metaschema-java/blob/main/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/metaschema-framework/metaschema-java/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website and readme documentation affected by the changes you made?

### CODE REVIEW

**Comment by david-waltermire** [comment]
This is broken, since it depends on metaschema-framework/metaschema#41

**Review by aj-stein-gsa — COMMENTED** [review]
Do we not need to worry about the code build failures in CI/CD? Other than that, amazing work!

**Comment by david-waltermire** [comment]
> metaschema-framework/metaschema#41

We do. I was waiting on approval of https://github.com/metaschema-framework/metaschema/pull/41 to fix this.

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**
